config/methods/common_methods.config

```diff
@@ -391,6 +391,63 @@ methods {
         process.containerOptions  = {-> (task.containsKey('cpus')) ? "--cpu-shares ${default_cpu_shares} --cpus ${task.cpus}" : "--cpu-shares ${default_cpu_shares}"}
     }
 
+    /**
+     * Configure all processes to save their command files in the output
+     * directory.
+     *
+     * This add a custom process directive that, when used as the afterScript,
+     * will copy all of the process's .command.* files into the output
+     * directory.
+
+     * Processes can customize the output directory by setting
+     * `process.ext.log_dir` and `process.ext.log_dir_suffix`. Both may be
+     * closures.
+     *
+     * Inspired by https://github.com/nextflow-io/nextflow/issues/1166#issuecomment-502467562
+     */
+    setup_process_afterscript = {
+        process.ext.log_dir = {
+            "${task.process.replace(':', '/')}"
+        }
+
+        process.ext.commonAfterScript = {
+            process_log_dir = [
+                "${params.log_output_dir}",
+                "process-log",
+                "${task.ext.log_dir}${task.ext.log_dir_suffix ?: ''}"
+            ].join("/")
+
+            // Handle relative paths
+            if (process_log_dir.substring(0, 1) != "/") {
+                process_log_dir = "${launchDir}/${process_log_dir}"
+            }
+
+            return """\
+                mkdir -p "${process_log_dir}"
+                for filename in .command.*; do
+                    cp "\${filename}" "${process_log_dir}/log\${filename}"
+                done
```

Comment:
There are some processes that don't produce `.command.*` files (example: processes using an `exec` block won't produce these files) and I believe this will cause an error for those since it'll find no files matching the pattern you're looking for

Reply:
Weird - if no files match the glob it will run through the loop once with `filename` set to `.command.*`. Per [BashPitfalls](http://mywiki.wooledge.org/BashPitfalls#line-57) I either need to test if the file exists with `-e` or set the `nullglob` shell option:

```bash
#!/bin/bash

echo "Original"

for filename in .command.*; do
    echo "Found $filename"
done

echo "Using -e"

for filename in .command.*; do
    [ -e "$filename" ] || continue
    echo "Found $filename"
done

echo "Using nullglob"

shopt -s nullglob

for filename in .command.*; do
    echo "Found $filename"
done
```

```console
$ ./test.sh
Original
Found .command.*
Using -e
Using nullglob
```

Reply:
Done with 6d1a3bd.